### PR TITLE
New post: What is an Insecure Direct Object Reference?

### PR DESCRIPTION
**Pull request by DennisTraub**
# Content Pull Request Checklist

Please check to make sure your content submission meets the following requirements by adding an `x` between the brackets:

- [x] The title of this pull request reflects the title of my content.
- [x] I have reviewed the [Content Review Checklist](https://github.com/build-on-aws/content/blob/main/CONTENT_REVIEW_CHECKLIST.md).
- [x] This pull request contains one piece of content (post or page).

Is this pull request linked to a Content Proposal or Idea Suggestion Issue? If so, please link to it below using the number or full link.

**By submitting this pull request, I confirm I own this content, have permission to use each image included in this content, and that it will be released under the [CC BY-SA 4.0 SA License](/LICENSE).**

### CODE REVIEW

File: posts/owasp-top-10-defined/04-what is-an-insecure-direct-object-reference/index.md [hunk 1]
```diff
@@ -0,0 +1,153 @@
+---
+title: What is an Insecure Direct Object Reference?
+description: An insecure direct object reference (IDOR) is another one of the OWASP Top 10 vulnerabilities. This article explains what it is and demonstrates a way to mitigate the risk.
+tags:
+  - owasp
+  - application-security
+authorGithubAlias: DennisTraub
+authorName: Dennis Traub
+date: tbd
```
Comment (lenadroid) [inline posts/owasp-top-10-defined/04-what is-an-insecure-direct-object-reference/index.md:9]: If you enter a dummy date in the format of YYYY-MM-DD, it'll be able to fix the validation to generate a preview link. You'll then be able to see what your post looks like on the site before merging

File: posts/owasp-top-10-defined/04-what is-an-insecure-direct-object-reference/index.md [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+An IDOR occurs when an application unintentionally exposes a reference to an internal implementation object, such as a record identifier in a database or a filename in a filesystem, revealing the actual identifier and format used in the backend storage.
+
+While IDOR itself may not directly constitute a security issue, it can enable attackers to mount enumeration attacks, taking advantage of the exposed format or pattern. This allows them to build a collection of valid identifiers based on the discovered format or pattern and test them against the application, trying to access associated objects. 
+
+In essence, IDOR vulnerabilities highlight the importance of properly securing object references and implementing robust access controls to prevent unauthorized access and potential enumeration attacks.
+
+## Impact
+
+The impact of an IDOR attack can be ...
```
Comment (lenadroid) [inline posts/owasp-top-10-defined/04-what is-an-insecure-direct-object-reference/index.md:38]: Recommend replacing ... with :

File: posts/owasp-top-10-defined/04-what is-an-insecure-direct-object-reference/index.md [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+
+## A simple example
+
+To demonstrate the discovery and exploitation of an IDOR, we'll be using a deployment of the [OWASP Juice Shop application](https://owasp.org/www-project-juice-shop/). This is an open source web application that is intentionally insecure. You can obtain the application yourself if you wish to recreate what I'm showing here or to play around on your own with other vulnerabilities. Please make sure to only install it in a lab environment that is isolated from production resources.
+
+> If you're interested in learning more about the OWASP Juice Shop, there is a [free book](https://pwning.owasp-juice.shop/) that explains the app and its vulnerabilities in more detail.
+
+After deploying the application, e.g. on an Amazon EC2 instance behind a load balancer, we can create a user, browse the products, and add them to our shopping basket:
+
+![A screenshot of the OWASP Juice Shop home page](images\juice-shop-home.png)
```
Comment (lenadroid) [inline posts/owasp-top-10-defined/04-what is-an-insecure-direct-object-reference/index.md:53]: The images are not pulling up. The issue is with the `\` within the URL, which should be`/` I believe

File: posts/owasp-top-10-defined/04-what is-an-insecure-direct-object-reference/index.md [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+
+Let's click on **Your Basket** in the menu bar. As you can see, my shopping basket is empty:
+
+![A screenshot of my empty basket in OWASP Juice Shop](images\juice-shop-basket.png)
```
Comment (lenadroid) [inline posts/owasp-top-10-defined/04-what is-an-insecure-direct-object-reference/index.md:57]: The images are not pulling up. The issue is with the \ within the URL, which should be/ I believe

File: posts/owasp-top-10-defined/04-what is-an-insecure-direct-object-reference/index.md (hunk 1 above)
Comment (DennisTraub) [reply posts/owasp-top-10-defined/04-what is-an-insecure-direct-object-reference/index.md:9]: Thanks, I'll keep that in mind

File: posts/owasp-top-10-defined/04-what is-an-insecure-direct-object-reference/index.md (hunk 4 above)
Comment (DennisTraub) [reply posts/owasp-top-10-defined/04-what is-an-insecure-direct-object-reference/index.md:57]: Windows messing paths up again. Thanks.

Comment (DennisTraub) [reply posts/owasp-top-10-defined/04-what is-an-insecure-direct-object-reference/index.md]: Fixed

File: posts/owasp-top-10-defined/04-what is-an-insecure-direct-object-reference/index.md (hunk 3 above)
Comment (DennisTraub) [reply posts/owasp-top-10-defined/04-what is-an-insecure-direct-object-reference/index.md]: Fixed

File: posts/owasp-top-10-defined/04-what is-an-insecure-direct-object-reference/index.md (hunk 2 above)
Comment (DennisTraub) [reply posts/owasp-top-10-defined/04-what is-an-insecure-direct-object-reference/index.md]: Done

**Review by DennisTraub — COMMENTED** [review]
Everything fixed according to reviewer recommendations

**Comment by lenadroid** [comment]
Thank you @DennisTraub! Almost ready. Looks like there was a space between "what" and "is" that blocks validation. Could you please replace the space with a dash in this file name? `04-what is-an-insecure-direct-object-reference`?

**Review by bketelsen — APPROVED** [review]
approved after url is fixed.
Nice article!

**Comment by lenadroid** [comment]
We can merge on Monday after the file name is fixed @cherbk @DennisTraub

**Comment by lenadroid** [comment]
> Looks like there was a space between "what" and "is" that blocks validation. Could you please replace the space with a dash in this file name? `04-what is-an-insecure-direct-object-reference`?

hey @DennisTraub, could you take a look?

**Comment by cobusbernard** [comment]
@DennisTraub would you mind recreating this pull request? I did a force push to your repo to fix the merge that happened. This is due to working directly on you `main` branch, and it would have re-included previous merges to the main content repo.

**Pull request closed without merging**